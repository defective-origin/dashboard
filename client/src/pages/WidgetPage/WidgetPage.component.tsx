import React from 'react'

// ---| core |---
import { cn } from 'tools'
import { useWidget } from 'api'
import { useParams } from 'router'

// ---| pages |---
import Page, { PageProps } from 'pages/Page'

// ---| screens |---
import Reviews from 'screens/views/Reviews'

// ---| components |---
import Text from 'components/Text'
import Block from 'components/Block'
import Section from 'components/Section'
import Actions from 'components/Actions'

// ---| self |---
import css from './WidgetPage.module.scss'
import PlaceholderCard from 'screens/cards/PlaceholderCard'

export type WidgetPageProps = PageProps

/**
 * Component description.
 *
 * How to use
 * @example
 * <WidgetPage />
 */
export function WidgetPage(props: WidgetPageProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.WidgetPage, className)
  const { id } = useParams()
  const widget = useWidget(id)

  // TODO: Logo
  // TODO: counters [installed, stars, reviews, rating]
  // TODO: security | Data safety | https://play.google.com/store/apps/datasafety?id=org.telegram.messenger

  return (
    <Page className={_className} name={widget.name} p='md' {...otherProps}>
      <Page.Content scroll='y' g='lg'>
        <Block v='x' justifies='space-between'>
          <Block>
            <Text v='h3' content={widget.name} />
            <Text v='h6' content='Author' />
          </Block>

          <Text v='h6' content='Version 0.0.0' />
        </Block>

        <Actions items={[
          { variant: 'button', content: 'use', v: 'outlined', color: 'success' },
          { variant: 'button', content: 'Storybook/Playground', v: 'outlined', color: 'success' },
          { variant: 'button', content: 'Releases', v: 'outlined', color: 'success' },
          { variant: 'button', start: 'share', content: 'share'},
          { variant: 'button', start: 'payments', content: 'buy' },
        ]} size='xs' g='xs' />

        <Text v='body2' content={widget.description} />

        <Section title='Screens' scroll='x'>
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
          <PlaceholderCard height={300} width={150} />
        </Section>

        <Reviews />

        {children}
      </Page.Content>

      <Page.RightAside scroll='y' p='md' g='lg'>
        <Section title='Releases' v='column'>
          <PlaceholderCard height={200} />
          <PlaceholderCard height={200} />
          <PlaceholderCard height={200} />
          <PlaceholderCard height={200} />
          <PlaceholderCard height={200} />
        </Section>
      </Page.RightAside>
    </Page>
  )
}

WidgetPage.displayName = 'WidgetPage'

export default WidgetPage
