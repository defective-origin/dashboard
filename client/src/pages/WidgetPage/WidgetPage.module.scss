@import 'src/Launcher/UILauncher/UILauncher.module';

:global {
  .dark {
    // --text-color: white;
  }

  .light {
    // --text-color: black;
  }
}

.WidgetPage {
}
