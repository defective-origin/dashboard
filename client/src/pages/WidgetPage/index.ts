/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './WidgetPage.component'

/* INJECT_EXPORT_PLACE */
export * from './WidgetPage.component'
