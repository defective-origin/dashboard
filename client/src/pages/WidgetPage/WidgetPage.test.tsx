// ---| tests |---
import { render } from 'tests'

// ---| self |---
import WidgetPage from './WidgetPage.component'

describe('[WidgetPage] component', () => {
  it('should render component', () => {
    const container = render(<WidgetPage />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
