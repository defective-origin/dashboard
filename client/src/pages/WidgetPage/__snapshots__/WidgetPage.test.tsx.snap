// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[WidgetPage] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="_WidgetPage_8c672c"
  />
</DocumentFragment>
`;
